bluesky/traffic/route.py

```diff
@@ -422,224 +480,114 @@ def addwpt(self, iac, name, wptype, lat, lon, alt=-999., spd=-999., afterwp="",
         # For safety
         self.nwp = len(self.wplat)
 
+        name = name.upper().strip()
+
+        wplat = lat
+        wplon = lon
+
         # Be default we trust, distrust needs to be earned
         wpok = True   # switch for waypoint check
 
         # Check if name already exists, if so add integer 01, 02, 03 etc.
-        appi    = 0 # appended integer to name starts at zero (=nothing)
-        wprtename = name.upper()  # wp name for in route
-        while self.wpname.count(wprtename)>0:
-            appi = appi+1
-            wprtename = name.upper()+"%02d"%appi
+        wprtename = Route.get_available_name(
+            self.wpname, name)
 
         # Select on wptype
-        # ORIGIN: Wptype is origin?
-        if wptype == self.orig:
-
-            if not (name == bs.traf.id[iac] + "ORIG"):   # published identifier
-                i = bs.navdb.getaptidx(name.upper().strip())
-                wpok = (i >= 0)
-                if wpok:
+        # ORIGIN: Wptype is origin/destination?
+        if wptype == Route.orig or wptype == Route.dest:
+            orig = wptype == Route.orig
+            wpidx = 0 if orig else -1
+            suffix = "ORIG" if orig else "DEST"
+
+            if not name == bs.traf.id[iac] + suffix:  # published identifier
+                i = bs.navdb.getaptidx(name)
+                if i >= 0:
                     wplat = bs.navdb.aptlat[i]
                     wplon = bs.navdb.aptlon[i]
-                else:
-                    wplat = lat
-                    wplon = lon
-                    wpok  = True
-            else:                                 # lat/lon type
-                wplat = lat
-                wplon = lon
-                wpok  = True
-
-            if wpok:
-                # Overwrite existing origin
-                if self.nwp > 0 and self.wptype[0] == self.orig:
-                    self.wpname[0] = wprtename
-                    self.wptype[0] = wptype
-                    self.wplat[0]  = wplat
-                    self.wplon[0]  = wplon
-                    self.wpalt[0]  = alt
-                    self.wpspd[0]  = spd
-                    self.wpflyby[0] = self.swflyby
-
-                # Or add before the first waypoint in route
-                else:
-                    self.wpname = [name.upper()] + self.wpname
-                    self.wptype = [wptype] + self.wptype
-                    self.wplat  = [wplat]  + self.wplat
-                    self.wplon  = [wplon]  + self.wplon
-                    self.wpalt  = [alt]  + self.wpalt
-                    self.wpspd  = [spd]  + self.wpspd
-                    self.wpflyby = [self.swflyby] + self.wpflyby
-
-                self.nwp    = self.nwp + 1

```

Comment:
Incrementing the number of wps whether inserting or overwriting is a bug.  This is corrected here (increment on insertion only)